packages/contracts/contracts/registry/InterfaceBasedRegistry.sol

```diff
@@ -64,16 +66,13 @@ abstract contract InterfaceBasedRegistry is UUPSUpgradeable, DaoAuthorizableUpgr
     /// @dev The managing DAO needs to grant REGISTER_PERMISSION_ID to registrar.
     /// @param _registrant The address of an [ERC-165](https://eips.ethereum.org/EIPS/eip-165) contract.
     function _register(address _registrant) internal {
-        if (!_registrant.isContract()) {
-            revert ContractAddressInvalid({registrant: _registrant});
+        if (entries[_registrant]) {
+            revert ContractAlreadyRegistered({registrant: _registrant});
         }
 
-        if (entries[_registrant]) revert ContractAlreadyRegistered({registrant: _registrant});
-
-        try IERC165Upgradeable(_registrant).supportsInterface(targetInterfaceId) returns (bool result) {
-            if (!result) revert ContractInterfaceInvalid(_registrant);
-        } catch {
-            revert ContractERC165SupportInvalid({registrant: _registrant});
+        // Will revert if address is not a contract or doesn't fully support targetInterfaceId + ERC165.
+        if (!_registrant.supportsInterface(targetInterfaceId)) {
+            revert ContractInterfaceInvalid(_registrant);
```

Comment:
`InvalidContractInterface` ?

Reply:
Let's leave it. I did as you. saind in `PluginRePo`, but seems like the code base follows `ContractInterfaceInvalid` practice more than  `InvalidContractInterface`.